### PR TITLE
Exercise 1 Clear

### PR DESCRIPTION
**Pull request by ParkJH1**
Code Review Plz

### CODE REVIEW

File: web/src/components/itemList.js [hunk 1]
```diff
@@ -19,27 +19,32 @@ class ItemList extends Component {
...
+    itemAdd = event => {
         const idx = event.target.getAttribute('id');
-        if(this.selected[idx]) {
-            this.props.removeItem(this.state.items[idx].price);
-            event.target.setAttribute('variant', 'primary')
-        }else {
-            this.props.addItems(this.state.items[idx].price);
-            event.target.setAttribute('variant', 'secondary')
-        }
+        var amt = parseInt(this.amount[idx].value);
```
Comment (c0510gy) [inline web/src/components/itemList.js]: Don't use `var` use `let` or `const`

File: web/src/components/itemList.js [hunk 2]
```diff
@@ -19,27 +19,32 @@ class ItemList extends Component {
...
+        const items = this.state.items;
```
Comment (c0510gy) [inline web/src/components/itemList.js]: What is this for?

File: web/src/components/itemList.js [hunk 3]
```diff
@@ -19,27 +19,32 @@ class ItemList extends Component {
...
+        if(isNaN(amt)) alert("개수는 숫자로 입력하세요")
+        else if(amt < 0) alert("최소 1개 이상 담아야 합니다")
+        else this.props.addItems(this.state.items[idx].price, amt, idx);
+    }
 
-        this.selected[idx] = !this.selected[idx];
+    itemRemove = event => {
+        const idx = event.target.getAttribute('id');
+        var amt = parseInt(this.amount[idx].value);
```
Comment (c0510gy) [inline web/src/components/itemList.js]: Again, don't use `var`

Fix same problem on other code

File: web/src/components/itemList.js [hunk 4]
```diff
@@ -19,27 +19,32 @@ class ItemList extends Component {
...
+        const items = this.state.items;
```
Comment (c0510gy) [inline web/src/components/itemList.js]: Again, what is this?

File: web/src/components/itemList.js [hunk 5]
```diff
@@ -19,27 +19,32 @@ class ItemList extends Component {
...
+        if(isNaN(amt)) alert("개수는 숫자로 입력하세요")
+        else if(amt < 0) alert("최소 1개 이상 담아야 합니다")
```
Comment (c0510gy) [inline web/src/components/itemList.js]: 담아 -> 빼야

File: web/src/components/itemList.js [hunk 6]
```diff
@@ -67,11 +72,16 @@ class ItemList extends Component {
...
-                    <Button variant={
-                        this.selected[i] ? 'secondary' : 'primary'
-                    } id={i} onClick={this.itemClicked}>{
-                        this.selected[i] ? 'Remove from cart' : 'Add to cart'
-                    }</Button>
+                    <InputGroup className="mb-3">
+                        {/* <Form.FormControl
+                            placeholder="Recipient's username"
+                        /> */}
+                        <Form.Control type="number" id={i} placeholder="개수 입력" ref={(ref) => { this.amount[i] = ref; return true; }}/>
```
Comment (c0510gy) [inline web/src/components/itemList.js]: Why there is `return`? Don't need, remove it

File: web/src/pages/shoppingBasket.js [hunk 7]
```diff
@@ -9,22 +9,43 @@ class ShoppingBasket extends Component {
...
 
-    addItem = price => {
+    addItem = (price, amount, idx) => {
+        var amountOfItems = this.state.amountOfItems
+        var sumOfAmounts = this.state.sumOfAmounts
+        var totalPrice = this.state.totalPrice
+        if(isNaN(parseInt(amountOfItems[idx]))) amountOfItems[idx] = 0
+        sumOfAmounts -= amountOfItems[idx]
+        totalPrice -= amountOfItems[idx] * price
+        amountOfItems[idx] = amountOfItems[idx] + amount
```
Comment (c0510gy) [inline web/src/pages/shoppingBasket.js]: Use `+=`

File: web/src/pages/shoppingBasket.js [hunk 8]
```diff
@@ -9,22 +9,43 @@ class ShoppingBasket extends Component {
...
-            numberOfSelectedItems: 0,
+            amountOfItems: [],
+            sumOfAmounts: 0,
             totalPrice: 0,
         };
     }
 
-    addItem = price => {
+    addItem = (price, amount, idx) => {
+        var amountOfItems = this.state.amountOfItems
```
Comment (c0510gy) [inline web/src/pages/shoppingBasket.js]: Use `const`

Also, it doesn't copy the array. You have to [treat `this.state` as if it were immutable](https://reactjs.org/docs/react-component.html). That means you have to deep copy an array whenever you want to update the array in the state.

You can deep copy the array like this: `const newAmountOfItems = [...this.state.amountOfItems];`

Same problem on removeItem

File: web/src/pages/shoppingBasket.js [hunk 9]
```diff
@@ -9,22 +9,43 @@ class ShoppingBasket extends Component {
...
+        var sumOfAmounts = this.state.sumOfAmounts
+        var totalPrice = this.state.totalPrice
+        if(isNaN(parseInt(amountOfItems[idx]))) amountOfItems[idx] = 0
+        sumOfAmounts -= amountOfItems[idx]
```
Comment (c0510gy) [inline web/src/pages/shoppingBasket.js]: Isn't `sumOfAmounts += amount` much more easier to understand?

Same problem on removeItem

**Review by c0510gy — CHANGES_REQUESTED** [review]

File: web/src/components/itemList.js (hunk 2 above)
Comment (ParkJH1) [reply web/src/components/itemList.js]: Done in 958e1b81d1003bf87de1114dc6525fda874bff66

File: web/src/components/itemList.js (hunk 4 above)
Comment (ParkJH1) [reply web/src/components/itemList.js]: Done in 958e1b81d1003bf87de1114dc6525fda874bff66

File: web/src/components/itemList.js (hunk 6 above)
Comment (ParkJH1) [reply web/src/components/itemList.js]: Done in 88ef4995b36b734c6ed87f0a1c7ae88f3caac5e7

File: web/src/components/itemList.js (hunk 5 above)
Comment (ParkJH1) [reply web/src/components/itemList.js]: Done in bcd008a3406ece1778aef2bf0690566f854d696c

File: web/src/pages/shoppingBasket.js (hunk 7 above)
Comment (ParkJH1) [reply web/src/pages/shoppingBasket.js]: Done in 56d186b9346970ea0250703502c4182b3e89925f

File: web/src/pages/shoppingBasket.js (hunk 9 above)
Comment (ParkJH1) [reply web/src/pages/shoppingBasket.js]: Done in 39b67c40ae66c9ec7104115a71ba082a13a815d9

File: web/src/pages/shoppingBasket.js (hunk 8 above)
Comment (ParkJH1) [reply web/src/pages/shoppingBasket.js]: Done in 0b4ef28f5351dd8b0222be5fc875319e8d3bf55f

**Review by ParkJH1 — COMMENTED** [review]
Done

**Pull request merged**